lib/services/resource.go

```diff
@@ -638,3 +640,41 @@ func (u *UnknownResource) UnmarshalJSON(raw []byte) error {
 	copy(u.Raw, raw)
 	return nil
 }
+
+type bootstrapResourcesType struct {
+	LocalResources  []types.Resource
+	DeviceResources []types.DeviceV1
+}
+
+func FilterBootstrapResourcesByType(resources ...types.Resource) (filteredResources bootstrapResourcesType) {
```

Comment:
Move this to lib/auth and make it private, there's little reason to have it here.